woocommerce/woocommerce-admin
# update i18n scripts for making JSON files

Fixes #3298

This PR addresses a number of issues with the existing `i18n` scripts and instructions. A`.pot` file is formatted with 3 sections per original string

- References: location(s) in the source where the string was found
- Original(s): the original untranslated string(s)
- Translation(s): one or more translations depending on the translation function called

```
#: includes/connect-existing-pages.php:33 src/Features/Analytics.php:142
#: client/header/activity-panel/panels/orders.js:198
#: client/layout/controller.js:64
msgid "Settings"
msgstr "
```

## Issues addressed

1. The `pot-to-php` script converted the `.js` pot file to a PHP file. The final `woocommerce-admin.pot` file contained no `.js` references for `wp i18n make-json` resulting in the command reporting success while creating no `.json` files.
1. The `npm run i18n:js` command only scanned the `client` folder so the strings in the `packages` folder weren't included in the translation file.
1. The `js` pot file contains source references (which are useful for translators) while the `json` files are compiled based on the registered script handles of the `dist/*` files. Those references needed to be updated after translation but before making the `json` files.

### Detailed test instructions:

- Run `npm run i18n`
- Edit `languages/woocommerce-admin.pot` and add translations for a few strings that have both `.php` and `.js` references
- Save the file as `woocommerce-admin-xx_YY.po` where `xx_YY` is a locale other than `en_US`
- Run `npm run i18n:json`
- Copy `.mo` and `.json` files to the `wp-content/languages/plugins` folder
- Change your locale to match the translation
- Check the dashboard for the translated version of those strings using the original file references to determine where the strings should appear
- Review changes to updated readme files


### Changelog Note:

Dev: Update translation scripts and instructions for creating `.json` files.
